tests/installation/welcome.pm

```diff
@@ -18,36 +18,44 @@ use testapi;
 use utils 'ensure_fullscreen';
 
 sub run() {
-    my @welcome_tags = [qw(inst-welcome inst-welcome-confirm-self-update-server)];
-    ensure_fullscreen;
-    my $bootup_timeout = 500;
-    if (get_var("BETA")) {
-        assert_screen "inst-betawarning", $bootup_timeout;
-        send_key "ret";
-        assert_screen @welcome_tags, 10;
-    }
-    else {
-        assert_screen @welcome_tags, $bootup_timeout;
-    }
-    if (match_has_tag('inst-welcome-confirm-self-update-server')) {
-        wait_screen_change { send_key $cmd{ok} };
-        assert_screen 'inst-welcome';
-    }
+    while (1) {
```

Comment:
It's not so simple as you think
any of examples you mentioned can occur without specifying it in test
I see two options use loop or create not small condition tree